buildSrc/src/main/kotlin/pklGradlePluginTest.gradle.kts

```diff
@@ -87,11 +87,11 @@ tasks.addRule("Pattern: compatibilityTest[All|Releases|Latest|Candidate|Nightly|
   }
 }
 
-fun createCompatibilityTestTask(versionInfo: GradleVersionInfo): Task =
+fun createCompatibilityTestTask(versionInfo: GradleVersionInfo) =
   createCompatibilityTestTask(versionInfo.version, versionInfo.downloadUrl)
 
-fun createCompatibilityTestTask(version: String, downloadUrl: String): Task {
-  return tasks.create("compatibilityTest$version", Test::class.java) {
+fun createCompatibilityTestTask(version: String, downloadUrl: String): TaskProvider<Test> {
```

Comment:
Updated the type on the right-hand side to be `TaskProvider<Test>` otherwise I get an error in IntelliJ.